consensus/src/state_computer.rs

```diff
@@ -100,16 +109,12 @@ impl StateComputer for ExecutionProxy {
         let pre_commit_instant = Instant::now();
         let synchronizer = Arc::clone(&self.synchronizer);
 
-        let committable_blocks = payload_and_output_list
+        let committable_blocks = blocks
             .into_iter()
-            .map(|payload_and_output| {
+            .map(|executed_block| {
                 CommittableBlock::new(
-                    payload_and_output
-                        .0
-                        .into_iter()
-                        .map(Transaction::UserTransaction)
-                        .collect(),
-                    payload_and_output.1,
+                    Self::transactions_from_block(executed_block.block()),
```

Comment:
I don't like that we do this `transactions_from_block` in both `compute` and `commit`. Would be great if we can just do this once when we initialize the block.

Reply:
We'll likely be cleaning up all this code with the TCB refactor. I should grab some time with you and the rest of the storage folks next week.

Reply:
I don't like this either but we can't inject directly into block since it depends on whether its ancestor has reconfiguration or not